articles/manual/operations-functions.md

```diff
@@ -283,3 +248,119 @@ function SquareOperation(op : (Qubit => Unit)) : (Qubit => Unit) {
 
 In principle, the classical logic within `SquareOperation` could have been much more involved, but it is still isolated from the rest of an operation by the guarantees that the compiler can offer about functions.
 This approach will be used throughout the Q# standard library for expressing classical control flow in a way that can be readily used within quantum programs.
+
+## Generic Operations and Functions
+
+> [!TIP]
+> This section assumes some basic familiarity with [generics in C#](https://docs.microsoft.com/dotnet/csharp/programming-guide/generics/introduction-to-generics), [generics in F#](https://docs.microsoft.com/dotnet/fsharp/language-reference/generics/), [C++ templates](https://docs.microsoft.com/cpp/cpp/templates-cpp), or similar approaches to metaprogramming in other languages.
+
+Many functions and operations that we might wish to define do not actually heavily rely on the types of their inputs, but rather only implicitly use their types via some other function or operation.
+For example, consider the *map* concept common to many functional languages; given a function $f(x)$ and a collection of values $\{x_1, x_2, \dots, x_n\}$, map returns a new collection $\{f(x_1), f(x_2), \dots, f(x_n)\}$.
+To implement this in Q#, we can take advantage of first class functions.
+Let's write out a quick example of `Map`, using ★ as a placeholder while we figure out what types we need.
+
+```qsharp
+function Map(fn : ★ -> ★, values : ★[]) : ★[] {
```

Comment:
It looks like this example is still using pre-0.6 names, it might be good to change this to be `Mapped` to clearly indicate that it is a function, and to parallel the name of the library function [Microsoft.Quantum.Arrays.Mapped](https://docs.microsoft.com/qsharp/api/qsharp/microsoft.quantum.arrays.mapped).